crates/services/p2p/src/discovery.rs

```diff
@@ -184,81 +154,43 @@ impl NetworkBehaviour for DiscoveryBehaviour {
             }
         }
 
-        // poll Kademlia behaviour
-        while let Poll::Ready(kad_action) = self.kademlia.poll(cx, params) {
-            match kad_action {
-                NetworkBehaviourAction::GenerateEvent(
-                    KademliaEvent::UnroutablePeer { peer },
-                ) => {
-                    return Poll::Ready(NetworkBehaviourAction::GenerateEvent(
-                        DiscoveryEvent::UnroutablePeer(peer),
-                    ))
-                }
-
-                NetworkBehaviourAction::Dial { handler, opts } => {
-                    return Poll::Ready(NetworkBehaviourAction::Dial { handler, opts })
-                }
-                NetworkBehaviourAction::CloseConnection {
-                    peer_id,
-                    connection,
-                } => {
-                    return Poll::Ready(NetworkBehaviourAction::CloseConnection {
-                        peer_id,
-                        connection,
-                    })
-                }
-                NetworkBehaviourAction::NotifyHandler {
-                    peer_id,
-                    handler,
-                    event,
-                } => {
-                    return Poll::Ready(NetworkBehaviourAction::NotifyHandler {
-                        peer_id,
-                        handler,
-                        event,
-                    })
-                }
-                NetworkBehaviourAction::ReportObservedAddr { address, score } => {
-                    return Poll::Ready(NetworkBehaviourAction::ReportObservedAddr {
-                        address,
-                        score,
-                    })
-                }
-                _ => {}
-            }
-        }
-
+        // poll sub-behaviors
+        let mut poll_state = None;
+        if let Poll::Ready(kad_action) = self.kademlia.poll(cx, params) {
+            poll_state = Poll::Ready(kad_action).into();
```

Comment:
mdns can override the `poll_state` later. You can use `return` as it was before=) Events from mdns are optional, and the priority of events from Kademlia is higher=)

Reply:
Ugh. You're right. I just think `return` is terrible. K. I'll just do that.